src/System.Management.Automation/security/SecuritySupport.cs

```diff
@@ -1336,39 +1336,26 @@ public enum ResolutionPurpose
 
     internal static class AmsiUtils
     {
-        private static string GetProcessHostName(string processName)
-        {
-            return string.Concat("PowerShell_", processName, ".exe_0.0.0.0");
-        }
-
         internal static int Init()
         {
             Diagnostics.Assert(s_amsiContext == IntPtr.Zero, "Init should be called just once");
 
             lock (s_amsiLockObject)
             {
-                Process currentProcess = Process.GetCurrentProcess();
-                string hostname;
+                string appName;
                 try
                 {
-                    hostname = string.Concat("PowerShell_", Environment.ProcessPath, "_",
-                        currentProcess.MainModule.FileVersionInfo.ProductVersion);
-                }
-                catch (ComponentModel.Win32Exception)
-                {
-                    // This exception can be thrown during thread impersonation (Access Denied for process module access).
-                    hostname = GetProcessHostName(currentProcess.ProcessName);
+                    appName = string.Concat("PowerShell_", Environment.ProcessPath, "_", PSVersionInfo.ProductVersion);
                 }
-                catch (FileNotFoundException)
+                catch (Exception)
                 {
-                    // This exception can occur if the file is renamed or moved to some other folder
-                    // (This has occurred during Exchange set up).
-                    hostname = GetProcessHostName(currentProcess.ProcessName);
+                    Process currentProcess = Process.GetCurrentProcess();
+                    appName = string.Concat("PowerShell_", currentProcess.ProcessName, ".exe_", PSVersionInfo.ProductVersion);
```

Comment:
When there is an exception thrown from `appName = string.Concat("PowerShell_", Environment.ProcessPath, "_", PSVersionInfo.ProductVersion);`, it's more likely the exception is from `PSVersionInfo.ProductVersion`. Using `PSVersionInfo.ProductVersion` again in the catch block would likely result in another exception. Why not stick to the original code by using `.exe_0.0.0.0`?

Reply:
Ah, I saw the discussion about this at https://github.com/PowerShell/PowerShell/pull/15527#discussion_r649524700. I will reply in that discussion thread.